src/besselk.jl

```diff
@@ -431,3 +434,65 @@ end
 besselk_power_series_cutoff(nu, x::Float64) = x < 2.0 || nu > 1.6x - 1.0
 besselk_power_series_cutoff(nu, x::Float32) = x < 10.0f0 || nu > 1.65f0*x - 8.0f0
 
+
+"""
+  besselk_asymptoticexp(v, x, order)
+
+Computes the asymptotic expansion of K_ν w.r.t. argument. Accurate for large x, and faster than uniform asymptotic expansion for small to small-ish orders. The default order of the expansion in `Bessels.besselk` is 10.
+"""
+besselk_asexp_cutoff(nu, x::T) where T = (nu < 20.0) && (x > ASEXP_CUTOFF(T))
+
+# Compute the function for (_v, _v+1), where _v = v-floor(v), and then do
+# forward recursion. If the argument is really large, this should be faster than
+# the uniform asymptotic expansion and plenty accurate.
+function besselk_asymptoticexp(v, x::T, order) where T
+    _v  = v - floor(v)
+    (kv, kvp1) = _besselk_as_pair(_v, x, order)
+    v == _v && return kv
+    _v += one(v)
+    twodx = T(2)*inv(x)
+    while _v < v
+        (kv, kvp1) = (kvp1, muladd(kvp1, twodx*_v, kv))
+        _v += one(_v)
+    end
+    kvp1
+end
+
+# For now, no exponential improvement. It requires the exponential integral
+# function, which would either need to be lifted from SpecialFunctions.jl or
+# re-implemented. And with an order of, like, 10, this seems to be pretty
+# accurate and still faster than the uniform asymptotic expansion.
+function _besselk_as_pair(v, x::T, order) where{T}
+    twox  = T(2)
+    fv = 4*v*v
+    fvp1 = 4*(v+one(v))^2
+    _z = x
+    ser = zero(T)
+    ser_v = one(T)
+    ser_vp1 = one(T)
+    floatj = one(T)
+    ak_numv = fv   - floatj
+    ak_numvp1 = fvp1 - floatj
+    factj = one(T)
+    twofloatj = one(T)
+    eightj = T(8)
+    for _ in 1:order
+        # add to the series:
+        term_v = ak_numv/(factj*_z*eightj)
```

Comment:
We can avoid one of these divisions. @oscardssmith can the compiler recognize this repeat division? Not for sure if that would increase error though.

I'm not for sure if we can avoid this division completely? `_z` and `eightj` can be expressed as multiplications but `factj` I don't think so?

Reply:
Looks like it removes it for us.